src/simulator_core/entities/assets/ates_cluster.py

```diff
@@ -0,0 +1,167 @@
+            )
+        self.thermal_power_allocation = setpoints[PROPERTY_HEAT_DEMAND]
+        self.temperature_return = setpoints[PROPERTY_TEMPERATURE_RETURN]
+        self.temperature_supply = setpoints[PROPERTY_TEMPERATURE_SUPPLY]
+        adjusted_mass_flowrate = heat_demand_and_temperature_to_mass_flow(
+            self.thermal_power_allocation, self.temperature_supply, self.temperature_return
+        )
+
+        [self.temperature_supply, self.temperature_return] = self.run_rosim(adjusted_mass_flowrate,
+                                                                            self.temperature_supply,
+                                                                            self.temperature_return)
+
+        if adjusted_mass_flowrate > 0:
+            self.solver_asset.supply_temperature = self.temperature_return
+        else:
+            self.solver_asset.supply_temperature = self.temperature_supply
+        self.solver_asset.mass_flow_rate_set_point = adjusted_mass_flowrate  # type: ignore
+
+    def add_physical_data(self, esdl_asset: EsdlAssetObject) -> None:
+        """Method to add physical data to the asset.
+
+        :param EsdlAssetObject esdl_asset: The esdl asset object to add the physical data from.
+         :return:
+        """
+
+    def write_to_output(self) -> None:
+        """Placeholder to write the asset to the output.
+
+        The output list is a list of dictionaries, where each dictionary
+        represents the output of its asset for a specific timestep.
+        """
+        output_dict = {
+            PROPERTY_MASSFLOW: self.solver_asset.get_mass_flow_rate(1),
+            PROPERTY_PRESSURE_SUPPLY: self.solver_asset.get_pressure(0),
+            PROPERTY_PRESSURE_RETURN: self.solver_asset.get_pressure(1),
+            PROPERTY_TEMPERATURE_SUPPLY: self.solver_asset.get_temperature(0),
+            PROPERTY_TEMPERATURE_RETURN: self.solver_asset.get_temperature(1),
+        }
+        self.output.append(output_dict)
+
+    def init_rosim(self) -> None:
+        """Function to initailized Rosim from XML file."""
+        RosimSequential = autoclass('tno.calc.RosimSequential')
+
+        self.rosimObj = RosimSequential(xmlfilejava, False, 2)
+
+    def run_rosim(self, mass_flow_rate: float, temperature_supply: float,
+                  temperature_return: float) -> tuple[float, float]:
+        """Function to calculate storage temperature after injection and production.
+
+        :param float mass_flow_rate: mass flow rate going to (positive) / going out (negative)
+        to ATES [kg/s]
+        :param float temperature_supply: temperature in the hot pipe of ATES [K]
+        :param float temperature_return: temperature in the cold pipe of ATES [K]
+        """
+        volume_flow = mass_flow_rate * 3600 / 1000
+        timestep = 1  # HARDCODED to 1 hour
+
+        ates_flow = [volume_flow, -1 * volume_flow]
```

Comment:
i will add comment

Reply:
Still need to add some more comments, since now it is sometimes not clear why you are doing stuff, like adding 8 to the NZ, ect. This should be explained

Reply:
additional comment added, regarding ROSIM input manipulation I need to ask Dorien. It could be this is due to rosim solver assumptions